builder/dockerignore.go

```diff
@@ -26,8 +26,8 @@ type DockerIgnoreContext struct {
 //
 // TODO: Don't require a ModifiableContext (use Context instead) and don't remove
 // files, instead handle a list of files to be excluded from the context.
-func (c DockerIgnoreContext) Process(filesToRemove []string) error {
-	f, err := c.Open(".dockerignore")
+func (c DockerIgnoreContext) Process(ignoreFile string, filesToRemove []string) error {
+	f, err := c.Open(ignoreFile)
 	// Note that a missing .dockerignore file isn't treated as an error
```

Comment:
Good one; yes (without looking at how to accomplish it), I think it someone _specified_ a `.dockerignore` file, and it doesn't exist, that should be an error. And if no file was specified, it should not be an error.

But, now sure if that can be achieved :blush: